pkg/generators/servers.go

```diff
@@ -330,19 +331,60 @@ func (g *ServersGenerator) generateServerAdapterSource(resource *concepts.Resour
 			{{ $requestBodyParameters := requestBodyParameters . }}
 			{{ $requestBodyLen := len $requestBodyParameters }}
 			{{ $responseParameters := responseParameters . }}
+			{{ $requestQueryParameters := requestQueryParameters . }}
 	
 			func (a *{{ $adapterName }}) read{{ $requestName }}(r *http.Request) (*{{ $requestName }}, error) {
+				var err error
 				result := new({{ $requestName }})
-				result.query = r.Form
+				result.query = r.URL.Query()
 				result.path = r.URL.Path
 
+				{{ if $requestQueryParameters }}
+					{{ range  $requestQueryParameters }}
+						{{ $fieldName := fieldName . }}
+						{{ $fieldType := fieldType . }}
+						{{ $fieldName }}String := result.query.Get("{{ $fieldName }}")
```

Comment:
I think it would be better to put this parsing logic in methods inside the `readers.go` file that is generated by the `generateHelpers` function. They could be functions like these:

```go
func RedInteger(s string) (value *int, err error) {
        ...
}

func ReadFloat(s string) (value *float, err error) {
        ...
}

...
``` 

Then this code would only need to call those functions:

```
result.{{ $fieldName }}, err = helpers.Read{{ $fieldType }}(query.Get("{{ $fieldName }}")
if err != nil {
        ...
}
```

You may also want to consider to pass to those functions the complete query object and the name of the parameter, so the logic to generate nice error messages (including the name of the parameter) can be inside them.

Remember to handle `Date` as well.

Reply:
Sounds good, will do.

Reply:
@jhernand @cben PTAL